crates/red_knot_python_semantic/src/types.rs

```diff
@@ -756,6 +756,8 @@ impl<'db> Type<'db> {
                         },
                     )
             }
+            (Type::Never, Type::Never) => false,
+            (_, Type::Never) => false,
```

Comment:
And I think this case should also not be necessary, because we fallback to `is_subtype_of`, which already contains this same case for `Never`. Removing this line and the previous one, all tests still pass.
```suggestion
```